components/eam/src/chemistry/modal_aero/aero_model.F90

```diff
@@ -1599,110 +1599,106 @@ subroutine modal_aero_bcscavcoef_init
           call calc_1_impact_rate( &
-               dg0_cgs, sigmag, rhowetaero_cgs, temp, press, &
+               dg0_cgs, sigmag, rhowetaero_cgs, temp_0C, press_750hPa, &
                scavratenum, scavratevol, lunerr )
 
-          nnfit = nnfit + 1
-          if (nnfit .gt. nnfit_maxd) then
-             write(lunerr,9110)
-             call endrun()
-          endif
-9110      format( '*** subr. modal_aero_bcscavcoef_init -- nnfit too big' )
-
-          xxfitnum(1,nnfit) = 1._r8
-          yyfitnum(nnfit) = log( scavratenum )
-
-          xxfitvol(1,nnfit) = 1._r8
-          yyfitvol(nnfit) = log( scavratevol )
-
-          scavimptblnum(jgrow,mode) = yyfitnum(1)
-          scavimptblvol(jgrow,mode) = yyfitvol(1)
+          scavimptblnum(jgrow,imode) = log( scavratenum )
+          scavimptblvol(jgrow,imode) = log( scavratevol )
 
        enddo growloop
     enddo modeloop
     return
+
   end subroutine modal_aero_bcscavcoef_init
 
   !===============================================================================
-  subroutine modal_aero_bcscavcoef_get( m, ncol, isprx, dgn_awet, scavcoefnum, scavcoefvol )
+  subroutine modal_aero_bcscavcoef_get( imode, ncol, isprx, dgn_awet, & ! in
+                                        scavcoefnum, scavcoefvol      ) ! out
+    !-----------------------------------------------------------------------
+    ! compute impaction scavenging removal amount for aerosol volume and number
+    !-----------------------------------------------------------------------
 
     use modal_aero_data, only: dgnum_amode
-    !-----------------------------------------------------------------------
+
     implicit none
 
-    integer,intent(in) :: m, ncol
-    logical,intent(in):: isprx(pcols,pver)
-    real(r8), intent(in) :: dgn_awet(pcols,pver,ntot_amode)
-    real(r8), intent(out) :: scavcoefnum(pcols,pver), scavcoefvol(pcols,pver)
+    integer,  intent(in) :: imode, ncol
+    logical,  intent(in) :: isprx(pcols,pver)           ! if there is precip
+    real(r8), intent(in) :: dgn_awet(pcols,pver,ntot_amode)  ! wet aerosol diameter [m]
+    real(r8), intent(out):: scavcoefnum(pcols,pver)     ! scavenging removal for aerosol number [1/h]
+    real(r8), intent(out):: scavcoefvol(pcols,pver)     ! scavenging removal for aerosol volume [1/h]
 
-    integer i, k, jgrow
-    real(r8) dumdgratio, xgrow, dumfhi, dumflo, scavimpvol, scavimpnum
+    ! local variables
+    integer  :: icol, kk, jgrow         ! index
+    real(r8) :: wetdiaratio             ! ratio of wet and dry aerosol diameter [fraction]
+    real(r8) :: xgrow, dumfhi, dumflo   ! working variables
+    real(r8) :: scavimpvol, scavimpnum  ! log of scavenging rates for volume and number
 
 
-    do k = 1, pver
-       do i = 1, ncol
+    do kk = 1, pver
+       do icol = 1, ncol
 
-          ! do only if no precip
-          if ( isprx(i,k) ) then
-             !
+          ! do only if there is precip
```

Comment:
So the previous comment is incorrect? I always worry about these things.

Reply:
yep, it's easy to check that `isprx` is just the logical check if precip>1e-7mm/hr, so .true. is precip